modules/porous_flow/doc/content/modules/porous_flow/heterogeneous_models.md

```diff
@@ -0,0 +1,97 @@
+
+In this example, the permeability in the data file is in millidarcys. As PorousFlow expects permeability
+in SI units of m$^2$, we multiply each permeability value by $9.869233 \times 10^{-16}$ and save these
+values in a new AuxVariable:
+
+!listing modules/porous_flow/examples/reservoir_model/regular_grid.i block=AuxKernels
+
+As this interpolation and multiplication is only required at the beginning of the simulation, we set
+the `execute_on` parameter to `initial` only.
+
+Finally, the heterogeneous permeability (in m$^2$) can be used in the calculation using a [PorousFlowPermeabilityConstFromVar](PorousFlowPermeabilityConstFromVar.md) material.
+
+!listing modules/porous_flow/examples/reservoir_model/regular_grid.i block=Materials/permeability
+
+The above steps create the following heterogeneous model that can then be used in a simulation:
+
+!media media/porous_flow/spe10_case1.png
+       id=fig:regular_grid
+       style=width:80%;margin-left:10px;
+       caption=Heterogeneous permeability for SPE comparative problem case 1
+
+## Reservoir models
+
+Often geological models are created using a modelling package to create realistic interpretations
+of the geology. In this case, some pre-processing of the geological model is often required before
+it can be used in PorousFlow.
+
+This pre-processing must be performed using some software that can take a geological model and convert
+it into a form that PorousFlow can read. One example of a pre-processing code that may be used to achieve
+this is [em2ex](https://github.com/cpgr/em2ex). This code converts a reservoir model in either Eclipse format
+or cell centered csv format to an Exodus mesh that can used in PorousFlow.
+
+The following example shows how to use an Exodus mesh that includes heterogeneity in a PorousFlow
+simulation. For this example, we use publicly available data from the [SAIGUP](https://www.nr.no/saigup)
+project to construct the heterogeneous reservoir model shown in [fig:field_model].
+
+!media media/porous_flow/saigup.png
+      id=fig:field_model
+      style=width:80%;margin-left:10px;
+      caption=Heterogeneous permeability for SAIGUP model
+
+The Eclipse reservoir model of the above structure is converted to an Exodus file using
+[em2ex](https://github.com/cpgr/em2ex). The heterogeneous porosity and permeability can
+then be read from the grid and used in the calculations using the following steps:
+
+First, the mesh (containing the heterogeneous reservoir properties) is read.
+
+!listing modules/porous_flow/examples/reservoir_model/field_model.i block=Mesh
+
+Constant monomial AuxVariables are then created. As the reservoir model again contains permeability
+in millidarcys, additional AuxVariables are also declared to hold the permeability in SI units (m$^2$).
+
+The AuxVariables for permeability in millidarcys are given initial values contained
+in the mesh file using the `initial_from_file_var` parameter, effectively transferring
```

Comment:
not sure about the word "effectively" here.  It suggests, to me, some sort of approximation.   Perhaps "... parameter, which transfers the heterogeneity..."